cli/sql: various fixes

knz opened the pull request:
See the individual commits for details.
Epic: CRDB-22182

Fixes #93138.
Fixes #92943.
Fixes #93345.

knz committed:
cli/sql: remove color styling of the line editor

Previously, we were using the default bubbline color defaults,
themselves inherited from the upstream textarea bubble.
On certain terminal / color theme combinations, these default
colors don't work: the background color for the prompt and
the cursor line becomes blurred with the foreground text,
or the text becomes too light for the terminal background.

Instead of researching a new color theme that would work for everyone,
this commit takes a shortcut and simply removes the color themes
altogether. This is "just" a SQL shell after all.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93343)
<!-- Reviewable:end -->

knz committed:
deps: update bubbline

This fixes the signal handling for SIGTERM.

Release note: None

knz committed:
cli/sql: the self-interrupt should send SIGINT not SIGHUP

This was a buglet left over from the first bubbline integration.

Release note: None

knz committed:
cli/demo: support `--pid-file`

Release note (cli change): `cockroach demo` now supports `--pid-file`
for symmetry with `cockroach start`.

knz committed:
cli/sql: properly handle SIGTERM

Bubbline returns an error upon SIGTERM, so we need to translate it
back into a signal.

Release note: None

ZhouXing19 reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> ! 

> Reviewed 1 of 1 files at r1, 5 of 5 files at r2, 1 of 1 files at r3, 3 of 3 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/93343#-:-NIsqxvK3MMt92iKOjz6:bks4l3i)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

knz commented:
TFYR!

bors r=ZhouXing19
